reconcile.go

```diff
@@ -234,54 +244,99 @@ func (a *Autopilot) pruneDeadServers() error {
 
 	state := a.GetState()
 
-	failed, voters, err := a.getFailedServers()
+	failed, vr, err := a.getFailedServers()
 	if err != nil || failed == nil {
 		return err
 	}
 
 	failed = a.promoter.FilterFailedServerRemovals(conf, state, failed)
 
-	// remove failed non voting servers
-	for _, srv := range failed.FailedNonVoters {
-		a.logger.Info("Attempting removal of failed server node", "id", srv.ID, "name", srv.Name, "address", srv.Address)
-		a.delegate.RemoveFailedServer(srv)
+	// Remove servers in order of increasing precedence
+	// Update the voter registry after each batch is processed
+
+	// remove stale non-voters
+	toRemove := a.adjudicateRemoval(failed.StaleNonVoters, vr)
```

Comment:
I guess in principle you could now collapse these 4 calls into 2, e.g.
```
stale := append(append([]raft.ServerID{}, failed.StaleNonVoters...), failed.StaleVoters...)
toRemove = a.adjudicateRemoval(stale, vr)
if err = a.removeStaleServers(toRemove); err != nil {
	return err
}
vr.removeAll(toRemove)
```

and similarly for the failed servers.  Up to you, it's also fine as you have it.